File: 0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        list1=[]
        for i in tokens:
            if i.lstrip('-').isdigit():
                list1.append(int(i))
            else:
                x=list1.pop()
                y=list1.pop()
                if i=='+':
                    list1.append(x+y)
                elif i=='-':
                    list1.append(y-x)
                elif i=='*':
                    list1.append(y*x)
                else:
                    list1.append(int(y/x))
        return list1[0]

File: 0150-evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> list=new Stack<>();
        for(String ch:tokens){ 
            if(ch.length()>1||Character.isDigit(ch.charAt(0))){
                list.push(Integer.parseInt(ch));
            }
            else{
                char c=ch.charAt(0);
                int x=list.pop();
                int y=list.pop();
                if(c=='+'){
                list.push(x+y); 
                } 
                else if(c=='-'){
                    list.push(y-x);
                }
                else if(c=='*'){
                    list.push(x*y);
                }
                else{
                    list.push(y/x);
                }
            }
            
        }
        return list.pop();
    }
}

